src/ecs/resources/input.rs

```diff
@@ -34,6 +34,27 @@ impl<'a> Iterator for PressedMouseButtonsIterator<'a> {
     }
 }
 
+/// This struct holds state information about input devices.
+///
+/// For example, if a key is pressed on the keyboard, this struct will record
+/// that the key is pressed until it is released again. Usage requires pumping input events through
+/// this struct, using the following code in the [handle_events] method of [State].
```

Comment:
I didn't even think about using reference links to methods in the documentation. Nice! 😄